modules/drivers/druid/src/metabase/driver/druid.clj

```diff
@@ -29,11 +30,18 @@
   [_ query]
   (druid.qp/mbql->native query))
 
+(defn- add-timeout-to-query [query timeout]
+  (let [parsed (if (string? query)
+                 (json/parse-string query keyword)
+                 query)]
+    (assoc-in parsed [:context :timeout] timeout)))
```

Comment:
I could remove all the new code you added and no tests would fail as a result which is a worrisome. 

Maybe mocking `druid.client/do-query-with-cancellation` with `with-redefs` and then "executing" a query by calling `execute-reducible-query` to make sure the (correct) timeout gets spliced in would be an easy way to test it

Reply:
I've rewritten my test to use this simpler method.